migration/idempotent/015-materialize_operations.sql

```diff
@@ -0,0 +1,167 @@
+-- returns the time stamp before which the materialization is valid (exclusive)
+-- it is important for this to be effecient: the span_start_time index guarantees
+-- effeciency in this case.
+CREATE OR REPLACE FUNCTION _ps_trace.materialize_watermark()
+RETURNS timestamptz
+STABLE
+AS
+$func$
+    SELECT parent_start_time + interval '10 minutes' --time_bucket returns the lower end of the bucket so add an interval
+    FROM _ps_trace.operation_materialization
+    ORDER BY parent_start_time DESC
+    LIMIT 1
+$func$
+LANGUAGE SQL;
+
+--This procedure updates the materialization
+CREATE OR REPLACE PROCEDURE _ps_trace.execute_materialize_operations(log_verbose boolean = false, log_timing boolean = true)
+AS $func$
```

Comment:
```suggestion
SET search_path = pg_catalog
LANGUAGE plpgsql VOLATILE AS 
$func$
```

Reply:
Let's keep the signature in the top of the function, so that it's more reader friendly and is easier to edit. (there is a corresponding edit at the bottom of the func)